networkit/cpp/centrality/TopHarmonicCloseness.cpp

```diff
@@ -93,12 +113,18 @@ void TopHarmonicCloseness::run() {
 }
 
 void TopHarmonicCloseness::runNBcut() {
+    tlx::d_ary_addressable_int_heap<node, 2, Aux::GreaterInVector<double>> prioQ{hCloseness};
```

Comment:
I don't think that moving this here is a good idea because doing so you allocate O(n) memory every time you run nbcut. If you just need to ensure that the heap is empty then call `clear`; this won't require to re-allocate the heap every time you run the algorithm.

Reply:
Right, fixed.